packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -1028,61 +1118,64 @@ class DefaultTextStyleTransition extends AnimatedWidget {
-/// [AnimatedBuilder] to limit rebuilds to only the [Text] widget. The current count
-/// is stored in a [ValueNotifier], which rebuilds the [AnimatedBuilder]'s contents
-/// when its value is changed.
+/// [AnimatedBuilder] to limit rebuilds to only the [Text] widget. The current
+/// count is stored in a [ValueNotifier], which rebuilds the [AnimatedBuilder]'s
+/// contents when its value is changed.
 ///
-/// ** See code in examples/api/lib/foundation/change_notifier/change_notifier.0.dart **
+/// ** See code in examples/api/lib/widgets/transitions/listenable_builder.1.dart **
 /// {@end-tool}
-/// {@endtemplate}
 ///
 /// See also:
 ///
+///  * [ListenableBuilder], which is the same, but is named more appropriately
+///    for a builder triggered on changes in [Listenable]s other than
+///    [Animation]s.
 ///  * [TweenAnimationBuilder], which animates a property to a target value
 ///    without requiring manual management of an [AnimationController].
-class AnimatedBuilder extends AnimatedWidget {
+class AnimatedBuilder extends ListenableBuilder {
   /// Creates an animated builder.
   ///
   /// The [animation] and [builder] arguments must not be null.
   const AnimatedBuilder({
     super.key,
     required Listenable animation,
-    required this.builder,
-    this.child,
-  }) : assert(animation != null),
-       assert(builder != null),
-       super(listenable: animation);
-
-  /// Called every time the animation changes value.
-  final TransitionBuilder builder;
+    required super.builder,
+    super.child,
+  }) : super(listenable: animation);
 
-  /// The child widget to pass to the [builder].
-  ///
-  /// If a [builder] callback's return value contains a subtree that does not
-  /// depend on the animation, it's more efficient to build that subtree once
-  /// instead of rebuilding it on every animation tick.
+  /// The animation supplied to the constructor.
   ///
-  /// If the pre-built subtree is passed as the [child] parameter, the
-  /// [AnimatedBuilder] will pass it back to the [builder] function so that it
-  /// can be incorporated into the build.
+  /// Also accessible through the [listenable] getter.
+  Listenable get animation => super.listenable;
+
+  // Overridden getters to replace with documentation tailored to
+  // AnimatedBuilder.
+
+  /// Called every time the [animation] notifies about a change.
   ///
-  /// Using this pre-built child is entirely optional, but can improve
-  /// performance significantly in some cases and is therefore a good practice.
-  final Widget? child;
+  /// The child given to the builder should typically be part of the returned
+  /// widget tree.
+  @override
+  TransitionBuilder get builder => super.builder;
 
+  /// The child widget to pass to the [builder].
+  ///
+  /// {@macro flutter.widgets.transitions.ListenableBuilder.optimizations}
   @override
-  Widget build(BuildContext context) {
-    return builder(context, child);
-  }
+  Widget? get child => super.child;
```

Comment:
This override is unnecessary, I think? It isn't adding anything new over the super class.

Reply:
Yes, you're correct, I removed it.